app/Dockerfile

```diff
@@ -33,5 +61,73 @@ ENV HOST=0.0.0.0
 # Install application dependencies.
 # https://python-poetry.org/docs/basic-usage/#installing-dependencies
 RUN poetry install
+
 # Run the application.
 CMD ["poetry", "run", "python", "-m", "api"]
+
+#-------------------------------------------------------------------------------
+# Non-development build environment
+#-------------------------------------------------------------------------------
+FROM build as app-build
+
+# Runtime arguement for what user will the bulk of this stage run as
+ARG RUN_USER
+
+# In between ARG RUN_USER and USER ${RUN_USER}, the user is still root
+# If there is anything that needs to be ran as root, this is the spot
+
+# Change the user from root to ${RUN_USER}
+USER ${RUN_USER}
+
+# Set the application working directory.
+WORKDIR /app
+
+# Install dependencies
+# Use a project-local virtualenv to be easy to find
+ENV POETRY_VIRTUALENVS_IN_PROJECT true
+COPY . /app
+RUN poetry install --no-root --without dev
+
+# Install application dependencies
+# The name of the wheel file comes from pyproject.toml
+RUN poetry build --format wheel && poetry run pip install 'dist/template_application_flask-0.1.0-py3-none-any.whl'
```

Comment:
Is this the command that actually builds the `run-api` binary in `/app/.venv/bin`? I can't seem to figure it out for certain, but if I'm inferring correctly, having poetry build the python wheel is what actually creates the binaries that are defined in pyproject.toml. If my understanding is correct can we make the comment reflect that?
```suggestion
# Build the application binary (python wheel) defined in pyproject.toml
RUN poetry build --format wheel && poetry run pip install 'dist/template_application_flask-0.1.0-py3-none-any.whl'
```
I think the "# install application dependencies" comment belongs with the previous command

Reply:
Yes, this is that command that builds in `app/.venv` and what has the binaries. And I like that, it is more clear, in the next commit